### PR TITLE
Fixed issues #34, #35 and #36 mentioned by Nathanael

### LINKED ISSUE #34: Issues I found during Data Sanity
If an issue has a * that means I'm not sure if it's actually an issue or not, but it looked like it was. So if someone could looks at those and confirm or deny that they are issues, that would be great.

Aux is still in music
[`src/BusinessLayer/InternetReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/InternetReporting.cpp)
Line 126

Aux in music
[`src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Line 146
Line 337
Util::writeBoolsIntoArray also need to be changed, corresponding to changes made with music

Aux is in music, and Left and Right are switched
[`src/DataLayer/DriverControlsData.h`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/DataLayer/DriverControlsData.h)
Line 14: Aux is still in music
Line 10 and 11: Left and Right are switched
~~Looks as though driver inputs are missing~~ * Not an actual issue

Right and Left Switched
[`src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Lines 637 and 638

Right and left swapped
[`src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Line 331 and 332

Left and Right do not match communications protocol *
[`src/BusinessLayer/SerialReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/SerialReporting.cpp)
Lines 140 and 141

Missing Alive
[`src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Line 633
Therefore Line 641 should be Util::writeBoolsIntoArray(data, 2, lightsStatus, 6);

Offset for battery faults is off
[`src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Line 490: Should be Util::writeBoolsIntoArray(data, 4, limitFlagsArray, 16);

FOR BACK EMF IMAGINARY
Make sure you change the payload packet offsets if they are there

Back EMF Imaginary shouldn't exist
[`src/BusinessLayer/SerialReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/SerialReporting.cpp)
Line 114

Back EMF Imaginary shouldn't exist
[`src/DataLayer/MotorDetailsData.h`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/DataLayer/MotorDetailsData.h)
Line 13 and 30

Back EMF Imaginary shouldn't exist
[`src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Line 259 and 290

Spelling Error
[`src/BusinessLayer/SerialReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/SerialReporting.cpp)
Line 177 and 187: Should be motor, not mootor

Spelling Error
[`src/DataLayer/MotorFaultsData.h`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/DataLayer/MotorFaultsData.h)
Lines 9, 26, 45 and 63, should be motor, not mootor

Spelling Error
[`src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Lines 384, 394: Should be motor, not mootor

~~Wrong packet ID *
[`src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Line 644: should be 0x10~~
Resolved

### LINKED ISSUE #35: JSON Data Sanity
BackEMF vs BackEMFReal
[`/src/BusinessLayer/InternetReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/InternetReporting.cpp)
Lines 81, 100
Uses BackEMFReal but communications protocol has BackEMF
Fixed in PR #37 

Aux in music
[`/src/BusinessLayer/InternetReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/InternetReporting.cpp)
Line 126
Fixed in PR #37 

WeakPackFault points to voltageRedundancyFault instead of weakPackFault
[`/src/BusinessLayer/InternetReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/InternetReporting.cpp)
Line 205
Fixed in PR #37 

Back EMF Imaginary
Make sure you change the payload offsets if they are there
[`/src/BusinessLayer/SerialReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/SerialReporting.cpp)
Line 114
[`/src/DataLayer/MotorDetailsData.h`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/DataLayer/MotorDetailsData.h)
Lines 13, 32
[`/src/Tests/BusinessLayer/SerialReportingTest.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Lines 257, 290
Fixed in PR #37

### LINKED ISSUE #36: Spelling Error, should be motor not mootor
[`/src/BusinessLayer/InternetReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/InternetReporting.cpp)
Lines 151, 173

[`/src/BusinessLayer/InternetReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/BusinessLayer/SerialReporting.cpp)
Lines 177, 187

[`/src/DataTests/BusinessLayer/SerialReporting.cpp`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/Tests/BusinessLayer/SerialReportingTest.cpp)
Lines 384, 394

[`/src/DataLayer/MotorFault.h`](https://github.com/UCSolarCarTeam/Epsilon-Telemetry-Test-Tool/blob/master/src/DataLayer/MotorFaultsData.h)
Lines 9, 26, 45, 63

### PR DESCRIPTION
**Pull request by jtdkhamp**
Fixes #34, fixes #35, and fixes #36

### CODE REVIEW

File: src/BusinessLayer/SerialReporting.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace
 // Refer to https://docs.google.com/spreadsheets/d/1soVLjeD9Sl7z7Z6cYMyn1fmn-cG7tx_pfFDsvgkCqMU/edit#gid=0
 // These lengths only include the data. Not the checksum
     const int KEY_MOTOR_LENGTH = 43;
-    const int MOTOR_DETAILS_LENGTH = 69;
+    const int MOTOR_DETAILS_LENGTH = 65;
```
Comment (Reddragoneer1) [inline src/BusinessLayer/SerialReporting.cpp]: Why was this changed?

File: src/Tests/BusinessLayer/SerialReportingTest.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ namespace
     const unsigned int COBS_ADDITIONAL_FRAME_DATA_SIZE = 2;
 
     const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_KEY_MOTOR = 47;
-    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_MOTOR_DETAILS = 73;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_MOTOR_DETAILS = 69;
```
Comment (Reddragoneer1) [inline src/Tests/BusinessLayer/SerialReportingTest.cpp]: Why was this changed?

**Comment by Reddragoneer1** [comment]
Other then the comments I made, looks good! Great job catching the temperature spelling mistake, I totally missed that

**Comment by bill-luu** [comment]
Reference which issue that is being addressed in your description by adding "fixes #35". That will link the PR with the issue so that when the PR is closed, the issue is closed as well.  

Also, does this also address #34 as well?

File: src/BusinessLayer/InternetReporting.cpp [hunk 3]
```diff
@@ -202,7 +202,7 @@ QJsonObject InternetReporting::makeBatteryFaults()
     errorFlags.insert("OpenWiringFault", batteryFaultsData_.openWiringFault);
     errorFlags.insert("CurrentSensorFault", batteryFaultsData_.currentSensorFault);
     errorFlags.insert("PackVoltageSensorFault", batteryFaultsData_.packVoltageSensorFault);
-    errorFlags.insert("WeakPackFault", batteryFaultsData_.voltageRedundancyFault);
+    errorFlags.insert("WeakPackFault", batteryFaultsData_.weakPackFault);
```
Comment (enochtsang) [inline src/BusinessLayer/InternetReporting.cpp]: I don't recognize this fault, @bill-luu is this accurate?

File: src/Tests/BusinessLayer/SerialReportingTest.cpp [hunk 4]
```diff
@@ -640,16 +638,17 @@ TEST_F(SerialReportingTest, sendLightsTest) // TODO create function which build
...
     appendChecksum(data, payloadLength);
     // do some additional data checks
     ASSERT_THAT(data[0], Eq(0x0A)); // packet id
-    ASSERT_THAT(((data[1] & 0x01) == 0x01), lightsData_->lowBeams);
-    ASSERT_THAT(((data[1] & 0x02) == 0x02), lightsData_->highBeams);
-    ASSERT_THAT(((data[1] & 0x04) == 0x04), lightsData_->brakes);
-    ASSERT_THAT(((data[1] & 0x08) == 0x08), lightsData_->leftSignal);
-    ASSERT_THAT(((data[1] & 0x10) == 0x10), lightsData_->rightSignal);
-    ASSERT_THAT(((data[1] & 0x20) == 0x20), lightsData_->bmsStrobeLight);
+    ASSERT_THAT(((data[1] & 0x01) == 0x01), lightsData_->alive);
```
Comment (enochtsang) [inline src/Tests/BusinessLayer/SerialReportingTest.cpp]: Uh, this doesn't seem accurate to me, an alive packet wasn't recently added was it?

File: src/BusinessLayer/SerialReporting.cpp (hunk 1 above)
Comment (bill-luu) [reply src/BusinessLayer/SerialReporting.cpp]: Since we removed the imaginary stuff

File: src/Tests/BusinessLayer/SerialReportingTest.cpp (hunk 2 above)
Comment (bill-luu) [reply src/Tests/BusinessLayer/SerialReportingTest.cpp]: As above.

File: src/BusinessLayer/InternetReporting.cpp (hunk 3 above)
Comment (bill-luu) [reply src/BusinessLayer/InternetReporting.cpp]: Yep! 
image

File: src/Tests/BusinessLayer/SerialReportingTest.cpp (hunk 4 above)
Comment (bill-luu) [reply src/Tests/BusinessLayer/SerialReportingTest.cpp]: This diff is comparing to the wrong branch :/

**Pull request merged**